core/amber/src/main/python/core/runnables/data_processor.py

```diff
@@ -46,14 +75,14 @@ def process_tuple(self) -> None:
                 else:
                     port = port_id.id
 
-                output_iterator = (
-                    executor.process_tuple(tuple_, port)
-                    if isinstance(tuple_, Tuple)
-                    else executor.on_finish(port)
-                )
+                if isinstance(tuple_, Tuple):
+                    output_iterator = executor.process_tuple(tuple_, port)
+                else:
+                    self._set_output_state(executor.produce_state(port))
```

Comment:
I am against this. I thought we discussed it? process a tuple should NOT get a state as output.

Reply:
Fixed.